src/HDF5.jl

```diff
@@ -1752,23 +1752,32 @@ write(parent::Union{HDF5File, HDF5Group}, name::String, data::Union{T, AbstractA
 # For datasets, "write(dset, name, val)" means "a_write"
 write(parent::HDF5Dataset, name::String, data::Union{T, AbstractArray{T}}, plists...) where {T<:ScalarOrString} = a_write(parent, name, data, plists...)
 
-# Reading arrays using getindex: data = dset[:,:,10]
-function getindex(dset::HDF5Dataset, indices::Union{AbstractRange{Int},Int}...)
+
+# Indexing
+
+Array(dset::HDF5Dataset) = read(dset)
+
+Base.eachindex(::IndexLinear, A::HDF5Dataset) = Base.OneTo(length(A))
+Base.axes(dset::HDF5Dataset) = map(Base.OneTo, size(dset))
```

Comment:
I don't this this method is needed for this PR

Reply:
Its used in `to_indices` to convert `Colon` to range:

https://github.com/JuliaLang/julia/blob/master/base/indices.jl#L321

Reply:
strange mine was working without this definition